### DISCUSSION TITLE
Command line print opens SumatraPDF-Window

### DISCUSSION THREAD

**Discussion by WiSchOnline** [sumatrapdfreader/sumatrapdf#2484] [Q&A]
Hi.

When I print a pdf via command line, SumatraPDF opens the pdf in a window and doesn't close it.
It tried it with the options `-exit-when-done` and `-exit-on-print`, but the windows stays open?
The complete command line: `SumatraPDF.exe -print-to "<printer>" "<file>" -silent -exit-when-done`

Any suggestion, how I can print via command line without opening the pdf in the viewer?

**Comment by kjk** [comment]
Which version of sumatra?

**Reply by WiSchOnline** [reply]
3.3.3.0
Also tested older versions and 32bit instead of 64bit.

**Reply by GitHubRulesOK** [reply]
remove the -silent it only tends to hide the error message you need to see such as 

```
---------------------------
Printing problem.
---------------------------
Printer with given name doesn't exist
---------------------------
OK   
---------------------------

or

---------------------------
Error
---------------------------
Couldn't open file '<file>' for printing
---------------------------
OK   
---------------------------
```
this command does NOT open SumataPDF ONLY the portprompt for file name and if watching the spool queue it flows through in one instant without any SumataPDF window

`"c:\program files\SumataPDF\SumatraPDF.exe" -print-to "Microsoft XPS Class Driver" sandbox\toc.pdf`
